src/core/server/saved_objects/import/import_saved_objects.ts

```diff
@@ -50,35 +56,72 @@ export async function importSavedObjectsFromStream({
 
-  // Exit early if no objects to import
-  if (filteredObjects.length === 0) {
-    return {
-      success: errorAccumulator.length === 0,
-      successCount: 0,
-      ...(errorAccumulator.length ? { errors: errorAccumulator } : {}),
+  let objectsToCreate = validateReferencesResult.filteredObjects;
+  if (trueCopy) {
+    const regenerateIdsResult = regenerateIds(objectsFromStream);
+    importIdMap = regenerateIdsResult.importIdMap;
+  } else {
+    // Check single-namespace objects for conflicts in this namespace, and check multi-namespace objects for conflicts across all namespaces
+    const checkConflictsOptions = {
+      savedObjectsClient,
+      namespace,
+      ignoreRegularConflicts: overwrite,
     };
+    const checkConflictsResult = await checkConflicts(
+      validateReferencesResult.filteredObjects,
+      checkConflictsOptions
+    );
+    errorAccumulator = [...errorAccumulator, ...checkConflictsResult.errors];
+    importIdMap = new Map([...importIdMap, ...checkConflictsResult.importIdMap]);
+
+    // Check multi-namespace object types for origin conflicts in this namespace
+    const checkOriginConflictsOptions = {
+      savedObjectsClient,
+      typeRegistry,
+      namespace,
+      ignoreRegularConflicts: overwrite,
+      importIds: checkConflictsResult.importIds,
+    };
+    const checkOriginConflictsResult = await checkOriginConflicts(
+      checkConflictsResult.filteredObjects,
+      checkOriginConflictsOptions
+    );
+    errorAccumulator = [...errorAccumulator, ...checkOriginConflictsResult.errors];
+    importIdMap = new Map([...importIdMap, ...checkOriginConflictsResult.importIdMap]);
+    objectsToCreate = checkOriginConflictsResult.filteredObjects;
   }
 
   // Create objects in bulk
-  const bulkCreateResult = await savedObjectsClient.bulkCreate(filteredObjects, {
-    overwrite,
-    namespace,
-  });
-  errorAccumulator = [
-    ...errorAccumulator,
-    ...extractErrors(bulkCreateResult.saved_objects, filteredObjects),
-  ];
+  const createSavedObjectsOptions = { savedObjectsClient, importIdMap, overwrite, namespace };
+  const createSavedObjectsResult = await createSavedObjects(
+    objectsToCreate,
+    errorAccumulator,
+    createSavedObjectsOptions
+  );
+  errorAccumulator = [...errorAccumulator, ...createSavedObjectsResult.errors];
+
+  const successResults = createSavedObjectsResult.createdObjects.map(
+    ({ type, id, destinationId, originId }) => {
+      return {
+        type,
+        id,
+        ...(destinationId && { destinationId }),
+        ...(destinationId && !originId && !trueCopy && { trueCopy: true }),
+      };
+    }
+  );
 
   return {
+    successCount: createSavedObjectsResult.createdObjects.length,
     success: errorAccumulator.length === 0,
-    successCount: bulkCreateResult.saved_objects.filter((obj) => !obj.error).length,
-    ...(errorAccumulator.length ? { errors: errorAccumulator } : {}),
+    ...(successResults.length && { successResults }),
```

Comment:
`errors` already behaved this way, so I opted not to change it. I figured `successResults` should do the same.

Happy to change it if you think we should, not sure if returning an empty `errors` array would break any other potential consumers of this API.

Reply:
I don't think it's necessary to change